superset/db_engine_specs/hive.py

```diff
@@ -547,7 +551,10 @@ def update_impersonation_config(
 
     @staticmethod
     def execute(  # type: ignore
-        cursor, query: str, async_: bool = False
+        cursor,
+        query: str,
+        database_id: int,
```

Comment:
NIT: New field could have been added as last arg, but `db_engine_specs` isn't a public API so it doesn't matter. I think in any case we have to touchup all the classes that are overriding this method (like you did). I noticed that some overrides for this method define `**kwargs` and really don't think we should have different headers in derived classes, but that's probably outside the scope of this PR.

Reply:
I had to add here because only Hive has the `async_` parameter, which gets unpacked from `**kwargs` in the base class.